pkg/metrics/metrics.go

```diff
@@ -0,0 +1,181 @@
+// Package metrics provides functionality for recording and serving
+// metric information related to AWS API calls and internal driver operations.
+package metrics
+
+import (
+	"net/http"
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// Metric types
+	AWS metricType = iota
+	Internal
+)
+const (
+	// AWS API metrics
+	awsDurationMetric = "cloudprovider_aws_api_request_duration_seconds"
+	awsThrottleMetric = "cloudprovider_aws_api_throttled_requests_total"
+	awsErrorMetric    = "cloudprovider_aws_api_request_errors"
+
+	// Internal driver metrics
+	internalDurationMetric = "ebs_csi_aws_com_duration_seconds"
+	internalValueMetric    = "ebs_csi_aws_com_operation"
```

Comment:
So I know there was discussion about this above, but I want to bring it up again because I think it's still relevant in the latest version of the PR:

What is an "internal value metric"? I don't understand what you mean and I'm an EBS CSI Driver maintainer, imagine the user confusion.

This needs to be renamed to something that clearly indicates what it will be used for in a way that doesn't require reading the source code to find out.

Reply:
> This is just me spitballing, but I wonder if it would make sense for the caller to supply their own name of what the metric should be.

Thought about this initially but after deliberation figured one metric and distinguishing via the use of labels would be a cleaner solution. Can certainly rework this and push in that front if you recommend. FWIW we won't be stuffing CSI rpc or K8s API measurements under this metric -- will be exposed via `csi_operations_seconds` instead:

```
Container Storage Interface operation duration with gRPC error code status total
Type: Histogram
Labels: driver_name, grpc_status_code, method_name, migrated
```

https://kubernetes.io/docs/reference/instrumentation/metrics/

> This needs to be renamed to something that clearly indicates what it will be used for

Suggestions?

Reply:
Can we clarify what we intend to put under this metric as is? It's hard for me to make recommendations here because it's not clear to me what the `ebs_csi_aws_com_operation` metric will contain.